tools/accuracy_checker/custom_evaluators/colorization_evaluator.py

```diff
@@ -0,0 +1,220 @@
+            dataset_config['name']
+        )
+
+
+class BaseModel:
+    def __init__(self, network_info, launcher):
+        self.network_info = network_info
+        self.supported_format = ['.xml', '.bin']
+
+    def check_format(self, current_format):
+        if not os.path.splitext(current_format)[1] in self.supported_format:
+            raise ConfigError('{} format not supported'.format(self.supported_format))
+        return current_format
+
+    def predict(self, idenitifers, input_data):
+        raise NotImplementedError
+
+    def release(self):
+        pass
+
+
+class ColorizationTestModel(BaseModel):
+    def __init__(self, network_info, launcher):
+        super().__init__(network_info, launcher)
+        model_xml = super().check_format(str(network_info['test']['model']))
+        model_bin = super().check_format(str(network_info['test']['weights']))
+        self.color_coeff = str(network_info['test']['color_coeff'])
+        self.network = launcher.create_ie_network(model_xml, model_bin)
+        if not hasattr(launcher, 'plugin'):
+            launcher.create_ie_plugin()
+        self.exec_network = launcher.plugin.load(self.network)
+        self.input_blob = next(iter(self.network.inputs))
+        self.output_blob = next(iter(self.network.outputs))
+        self.test_mean = float(network_info['test']['color_mean'])
+
+    def predict(self, identifiers, input_data):
+
+        output_blob = next(iter(self.exec_network.outputs))
+        (h_orig, w_orig) = input_data[0].shape[:2]
+
+        new_result = []
+        for input in input_data:
+            input = input.astype(np.float32)
+            img_lab = cv2.cvtColor(input, cv2.COLOR_RGB2Lab)
+            img_l = img_lab[:, :, 0]
+
+            img_lab_rs = cv2.cvtColor(input, cv2.COLOR_RGB2Lab)
+            img_l_rs = img_lab_rs[:, :, 0]
+            img_l_rs -= self.test_mean
+
+            res = self.exec_network.infer(inputs={self.input_blob: [img_l_rs]})
+            color_coeff = np.load(self.color_coeff)
+            update_res = (res[output_blob] * color_coeff.transpose()[:, :, np.newaxis, np.newaxis]).sum(1)
+
+            out = update_res.transpose((1, 2, 0)).astype(np.float32)
+            out = cv2.resize(out, (w_orig, h_orig))
+            img_lab_out = np.concatenate((img_l[:, :, np.newaxis], out), axis=2)
+            self.img_bgr_out = np.clip(cv2.cvtColor(img_lab_out, cv2.COLOR_Lab2BGR), 0, 1)
+            new_result.append(self.img_bgr_out)
+        return new_result
+
+    def release(self):
+        del self.exec_network
+        del self.img_bgr_out
+
+    def fit_to_input(self, input_data):
+        input_data = np.reshape(input_data, self.network.inputs[self.input_blob].shape)
+        return {self.input_blob: input_data}
+
+
+class ColorizationCheckModel(BaseModel):
+    def __init__(self, network_info, launcher):
+        super().__init__(network_info, launcher)
+        model_xml = super().check_format(str(network_info['checker']['model']))
```

Comment:
the child inherits all methods from parent, no need to call this method from super(), just 
`self.check_format()`

Reply:
@eaidova corrected)